aptos-move/aptos-vm/src/data_cache.rs

```diff
@@ -48,7 +52,7 @@ pub struct StorageAdapter<'a, S> {
         RefCell<BTreeMap<AccountAddress, BTreeMap<StructTag, BTreeMap<StructTag, Vec<u8>>>>>,
 }
 
-impl<'a, S: StateView> StorageAdapter<'a, S> {
+impl<'a, S> StorageAdapter<'a, S> {
```

Comment:
Why is `StateView` trait bound removed here? Isn't StorageAdapter defined only for StateView?

Reply:
There are 2 impls: all functionality relies on state view, but one should be able to create an adapter on top of some other type and implement move resolver / state view on top.